device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/transport/IotHubTransport.java

```diff
@@ -337,6 +339,15 @@ public void onMultiplexedDeviceSessionLost(Throwable e, String connectionId, Str
         }
     }
 
+    @Override
+    public void onMultiplexedDeviceSessionRegistrationFailed(String connectionId, String deviceId, Exception e)
+    {
+        if (connectionId != null && connectionId.equals(this.iotHubTransportConnection.getConnectionId()))
```

Comment:
Q - "connectionId.equals(this.iotHubTransportConnection.getConnectionId())" is this to avoid the scenario where by the time this callback was fired the client had already initiated a new reconnection attempt?

Reply:
Yeah, the IotHubTransport layer may spawn a new AmqpsIotHubConnection instance during retry logic. This connectionId prevents us from logging/retrying errors from old connections that we've moved on from